staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go

```diff
@@ -917,20 +920,28 @@ func (t *Tester) testDeleteWithResourceVersion(obj runtime.Object, createFn Crea
 
 	foo := obj.DeepCopyObject()
 	t.setObjectMeta(foo, t.namer(1))
-	objectMeta := t.getObjectMetaOrFail(foo)
-	objectMeta.SetResourceVersion("RV0000")
 	if err := createFn(ctx, foo); err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
-	opts.Preconditions = metav1.NewRVDeletionPrecondition("RV1111").Preconditions
+	newObj, err := getFn(ctx, foo)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	objectMeta := t.getObjectMetaOrFail(newObj)
+
+	currentResourceVersion, err := strconv.Atoi(objectMeta.GetResourceVersion())
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
```

Comment:
no need to parse the current resource version and manipulate it numerically, can just set the precondition to some other string and the delete should fail

Reply:
@liggitt That way was suggested by @jingyih, @jingyih do you have any comment if I change it like this:

```go
opts.Preconditions = metav1.NewRVDeletionPrecondition("wrongVersion").Preconditions
```

Reply:
Sounds good to me. As long as there it is a clearly incorrect RV string.